### PR TITLE
Change the word 'number' to 'integer' in number field message

### PR DESCRIPTION
**Pull request by benebsiny**
The error messages of number fields are different, some use 'integer', the other use 'number'. For unity, it may be a good idea to change them to the same word 'integer'.

Besides, I deduplicated the code that prevent user to input 'E' or 'e'. See [here](https://github.com/jenkinsci/jenkins/pull/5145#pullrequestreview-560646203).

image

### Proposed changelog entries

* Change the word 'number' to 'integer' in the error message under the number field

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by res0nance** [comment]
The commit seems to only include the deduplication

**Comment by benebsiny** [comment]
@res0nance I changed the word of error message. I did not modified the validation regex because it had already can validate for 'integer'.

**Comment by res0nance** [comment]
> @res0nance I changed the word of error message. I did not modified the validation regex because it had already can validate for 'integer'.

Ah, you're right i failed to notice the change

**Review by res0nance — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
LGTM, thanks!

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Review by StefanSpieker — APPROVED** [review]

File: war/src/main/webapp/scripts/hudson-behavior.js [hunk 1]
```diff
@@ -670,6 +670,17 @@ function registerMinMaxValidator(e) {
     e = null; // avoid memory leak
 }
 
+
+/**
+ * Prevent user input 'e' or 'E' in <f:number>
```
Comment (daniel-beck) [inline war/src/main/webapp/scripts/hudson-behavior.js:675]: Does not prevent copy & paste?

**Comment by jtnord** [comment]
noting regression in ATH.
https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/master/335/testReport/core/FormValidationTest/java_8_split5___validate/

**Pull request merged**